dask/layers.py

```diff
@@ -64,43 +74,23 @@ def __dask_distributed_unpack__(cls, state):
         return cls(**state)
 
 
-class BlockwiseCreateArray(Blockwise):
-    """
-    Specialized Blockwise Layer for array creation routines.
+class ArrayChunkShapeDep(ArrayBlockwiseDep):
+    """Index-chunkshape mapping for array-like"""
 
-    Enables HighLevelGraph optimizations.
+    def __getitem__(self, idx: tuple):
+        return tuple(chunk[i] for i, chunk in zip(idx, self.chunks))
 
-    Parameters
-    ----------
-    name: string
-        The output name.
-    func : callable
-        Function to apply to populate individual blocks. This function should take
-        an iterable containing the dimensions of the given block.
-    shape: iterable
-        Iterable containing the overall shape of the array.
-    chunks: iterable
-        Iterable containing the chunk sizes along each dimension of array.
-    """
 
-    def __init__(
-        self,
-        name,
-        func,
-        shape,
-        chunks,
-    ):
-        # Define "blockwise" graph
-        dsk = {name: (func, blockwise_token(0))}
+class ArraySliceDep(ArrayBlockwiseDep):
+    """Index-slice mapping for array-like"""
```

Comment:
```suggestion
    "Produces the slice into the full-size array corresponding to the current chunk"
```
Maybe something like this would be a little easier to understand?